inc/namespace.php

```diff
@@ -17,6 +18,14 @@
 function bootstrap() {
 	add_action( 'muplugins_loaded', __NAMESPACE__ . '\\load_notifications', 0 );
 	add_action( 'muplugins_loaded', __NAMESPACE__ . '\\load_publication_checklist', 0 );
+	add_action( 'muplugins_loaded', __NAMESPACE__ . '\\load_duplicate_posts' );
+	add_action( 'admin_menu', __NAMESPACE__ . '\\remove_duplicate_post_admin_page', 99 );
+	add_filter( 'duplicate_post_enabled_post_types', __NAMESPACE__ . '\\maybe_override_enabled_post_types' );
+
+	$enabled_post_types = get_duplicate_post_types();
```

Comment:
Given this occurs so early it doesn't give custom code much chance to hook in - that is some code may want to add a filter to modify the post types but it isn't run until the `init` action has fired for example. You can use an action to defer this loop to the last possible moment for example:

```php
add_action( 'admin_init', __NAMESPACE__ . '\\filter_duplicate_post_columns', 1000 );

//...

function filter_duplicate_post_columns() {
	$enabled_post_types = get_duplicate_post_types();
	foreach ( $enabled_post_types as $post_type ) {
		add_filter( "manage_{$post_type}_posts_columns", __NAMESPACE__ . '\\remove_duplicate_post_original_item_column', 11 );
	}
}
```

I know a developer could maybe get around this by adding the modifications in a custom module but this is a kind of defensive coding in WordPress that helps avoid one of the problems that's hard to debug which is action / filter order.

Reply:
I like this solution anyway because I wasn't a huge fan of having a variable in the bootstrap function and this avoids that. :)